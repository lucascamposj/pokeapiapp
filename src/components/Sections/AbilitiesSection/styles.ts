import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  width: 100%;
  flex-direction: row;
  align-content: center;
  flex-wrap: wrap;
  justify-content: center;
`;
