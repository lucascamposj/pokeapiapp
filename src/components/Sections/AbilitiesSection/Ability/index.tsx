import React from 'react';
import { Container, AbilityText } from './styles';
import { AbilityInterface } from '../../../../pages/Pokemon';

const Ability: React.FC<AbilityInterface> = ({ ability }: AbilityInterface) => {
  return (
    <Container>
      <AbilityText>{ability.name}</AbilityText>
    </Container>
  );
};

export default Ability;
