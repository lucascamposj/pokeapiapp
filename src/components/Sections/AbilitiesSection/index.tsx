import React from 'react';
import Ability from './Ability';
import { Container } from './styles';
import { AbilityInterface } from '../../../pages/Pokemon';

interface AbilitySectionProps {
  abilities: AbilityInterface[] | undefined;
}

const AbilitySection: React.FC<AbilitySectionProps> = ({
  abilities,
}: AbilitySectionProps) => {
  return (
    <Container>
      {abilities?.map(({ ability }) => (
        <Ability key={ability.name} ability={ability} />
      ))}
    </Container>
  );
};

export default AbilitySection;
