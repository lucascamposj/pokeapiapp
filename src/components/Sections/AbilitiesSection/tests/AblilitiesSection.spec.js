import React from 'react';
import {shallow} from 'enzyme';
import AbilitiesSection from '../../AbilitiesSection';

describe('AbilitiesSection', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const abilities = [
              {
                ability: {
                  name: "Overgrow"
                }
              },
              {
                ability: {
                  name: "Chlorophyll"
                }
              }
            ]
            const component = shallow(
              <AbilitiesSection
                abilities={abilities}
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
