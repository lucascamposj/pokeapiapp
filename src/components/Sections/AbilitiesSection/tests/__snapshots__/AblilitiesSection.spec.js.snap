// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbilitiesSection Rendering should match to snapshot 1`] = `
<Styled(View)>
  <Ability
    ability={
      Object {
        "name": "Overgrow",
      }
    }
    key="Overgrow"
  />
  <Ability
    ability={
      Object {
        "name": "Chlorophyll",
      }
    }
    key="Chlorophyll"
  />
</Styled(View)>
`;
